exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1326 @@
+package bybit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/account"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/deposit"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/protocol"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// GetDefaultConfig returns a default exchange config
+func (by *Bybit) GetDefaultConfig() (*config.Exchange, error) {
+	by.SetDefaults()
+	exchCfg := new(config.Exchange)
+	exchCfg.Name = by.Name
+	exchCfg.HTTPTimeout = exchange.DefaultHTTPTimeout
+	exchCfg.BaseCurrencies = by.BaseCurrencies
+
+	err := by.SetupDefaults(exchCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	if by.Features.Supports.RESTCapabilities.AutoPairUpdates {
+		err := by.UpdateTradablePairs(context.TODO(), true)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return exchCfg, nil
+}
+
+// SetDefaults sets the basic defaults for Bybit
+func (by *Bybit) SetDefaults() {
+	by.Name = "Bybit"
+	by.Enabled = true
+	by.Verbose = true
+	by.API.CredentialsValidator.RequiresKey = true
+	by.API.CredentialsValidator.RequiresSecret = true
+
+	requestFmt := &currency.PairFormat{Uppercase: true}
+
+	configFmt := &currency.PairFormat{Uppercase: true}
+	err := by.SetGlobalPairsManager(requestFmt, configFmt, asset.Spot, asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures)
+	if err != nil {
+		log.Errorln(log.ExchangeSys, err)
+	}
+
+	err = by.DisableAssetWebsocketSupport(asset.CoinMarginedFutures)
```

Comment:
Websocket return handling: Have only ETH-USDT enabled in spot but this asset type is coming up as coinmargined futures and other issues.
[INFO] |ORDERBOOK| 07/03/2022 09:37:49 |Bybit websocket ETH-USDT COINMARGINEDFUTURES: ORDERBOOK: Bids len: 40 Amount: 137.025140 ETH. Total value: 361102.1280742 Asks len: 40 Amount: 147.589340 ETH. Total value: 389750.72931130003
[INFO] |TICKER| 07/03/2022 09:37:48 |Bybit websocket ICP-USDT SPOT: TICKER: Last 0.00000000 Ask 17.46000000 Bid 17.45000000 High 0.00000000 Low 0.00000000 Volume 0.00000000 (This should be a futures asset) etc.
[INFO] |TICKER| 07/03/2022 09:37:48 |Bybit websocket XEM-USDT SPOT: TICKER: Last 0.00000000 Ask 0.09828000 Bid 0.09784000 High 0.00000000 Low 0.00000000 Volume 0.00000000 (usdtmargined futures asset)